ncm-dpmlfc/src/main/perl/dpmlfc/RuleBasedEditor.pm

```diff
@@ -0,0 +1,817 @@
+# ${license-info}
+# ${developer-info}
+# ${author-info}
+# ${build-info}
+#
+#
+# This module implement a rule-based editor that is used to modify the content
+# of an existing file without taking care of the whole file. Each rule
+# driving the edition process is applied to one matching line. The input for
+# updating the file is the Quattor configuration and conditions can be defined
+# based on the contents of this configuration.
+#
+#######################################################################
+
+package NCM::Component::${project.artifactId}::RuleBasedEditor;
+
+use strict;
+use warnings;
+use NCM::Component;
+use vars qw(@ISA $EC);
+@ISA = qw(NCM::Component);
+$EC=LC::Exception::Context->new->will_store_all;
+
+use EDG::WP4::CCM::Element;
+
+use Readonly;
+
+use CAF::FileEditor;
+
+use Encode qw(encode_utf8);
+
+local(*DTA);
+
+# Constants use to format lines in configuration files
+# Exported constants
+use enum qw(LINE_FORMAT_PARAM=1
```

Comment:
Yes it must start at 1 for this one (it was the case before, I have not checked if there is a good reason but there is a risk that 0 is used a undefined).